src/user-form.js

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import { useState } from "react";
+
+function AuthPage(props) {
+    const {setUsername, setPassword} = props;
+    return (
+        <div style={{padding: "30px"}}>
+            <div><label style={{fontSize: "32px", marginTop: "25px"}}>User Name:
+                <input style={{fontSize: "24px", marginLeft: "10px"}} onChange={(e) => setUsername(e.target.value)}/>
+            </label></div>
+            <div><label style={{fontSize: "32px", marginTop: "25px"}}>Password:
+                <input style={{fontSize: "24px", marginLeft: "10px"}} onChange={(e) => setPassword(e.target.value)}/>
+            </label></div>
+        </div>
+    );
+}
+
+function CountryPage(props) {
+    const {country, setCountry} = props;
+
+    return (
+        <div>
+            <label style={{fontSize: "32px", marginTop: "25px"}}>Country
+                <select style={{fontSize: "24px"}} onChange={(e) => setCountry(e.target.value)}>
+                    {["US", "UK", "Israel", "Other"].map(x => <option>{x}</option>)}
+                </select>
+            </label>
+        </div>
+    );
+}
+
+function SummaryPage(props) {
+    const {username, password, country} = props;
+    return (
+        <div style={{padding: "30px"}}>
+            <div><label style={{fontSize: "32px", marginTop: "25px"}}>User Name: {username}</label></div>
+            <div><label style={{fontSize: "32px", marginTop: "25px"}}>Password: {password}</label></div>
+            <div><label style={{fontSize: "32px", marginTop: "25px"}}>Country: {country}</label></div>
+        </div>
+    );
+}
+
+
+export default function UserForm(props) {
+    const {title} = props;
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [country, setCountry] = useState("");
+    const [active, setActive] = useState(0)
+
+    function activePageIndex() {
+        return pages.map((page, i) => page.props.style.display === "block" ? i : -1).reduce((a, b) => a > b ? a : b);
```

Comment:
Waaaay overcomplicated! 🤯
I could not follow the logic of what you were trying to implement here.

The simple and straightforward solution would be to have an array to function as a map, and a state of the current index. Increment/decrement on next/prev clicks unless not possible.